telosnetwork/telos-wallet
# #604 | View allowances

# Fixes #604 

## Description
This PR enables viewing real allowance information in the UI (note that modifying allowances is not a part of it)

Note: the indexer does not currently return cancelled allowances, so the toggle for that is nonfunctional. The logic should be working but we will need to test that after the indexer is updated

Note: the indexer returns slightly malformed data for single ERC721 allowances, so they are not included in the table. they should appear when the indexer is updated

Note: in this PR, extra large allowances are called "huge". In the next PR this will be changed to "infinite" with additional logic

## Test scenarios
- run the code locally on mainnet: `git fetch --all && git checkout 604-view-allowances-api-integration && echo 'NETWORK="mainnet"' > .env && yarn dev`
- log in with the team account (`0x...eE37`)
- go to the allowances page
    - you should see a loading state, then-
    - you should see allowances sorted by asset quantity, descending
- use the pagination buttons at the bottom of the table to view the next page of allowances
    - you should see the next page of allowances
- change the rows per page value in the table controls
    - rows per page should change accordingly
- try various options for sorting via clicking (and using tab + space/enter) on the table headers
    - allowances should sort correctly
- type various search terms in the search bar
    - results should filter by search term (see the github issue for details on what should be filterable)

## Checklist:
